subsys/bluetooth/host/hci_core.c

```diff
@@ -3780,10 +3787,29 @@ int bt_disable(void)
 
 	bt_monitor_send(BT_MONITOR_CLOSE_INDEX, NULL, 0);
 
+#if defined(CONFIG_BT_EXT_ADV) || defined(CONFIG_BT_BROADCASTER)
+	bt_adv_reset_adv_pool();
+#endif /* CONFIG_BT_EXT_ADV || CONFIG_BT_BROADCASTER */
+
+#if defined(CONFIG_BT_PRIVACY)
+	k_work_cancel_delayable(&bt_dev.rpa_update);
+#endif /* CONFIG_BT_PRIVACY */
+
 #if defined(CONFIG_BT_PER_ADV_SYNC)
 	bt_periodic_sync_disable();
 #endif /* CONFIG_BT_PER_ADV_SYNC */
 
+#if defined(CONFIG_BT_CONN)
+#if defined(CONFIG_BT_SMP)
+	bt_pub_key_hci_disrupted();
+#endif /* CONFIG_BT_SMP */
+	bt_conn_cleanup_all();
+	disconnected_handles_reset();
+#endif /* CONFIG_BT_CONN */
+
+	err = bt_id_delete_all();
```

Comment:
This doesn't look correct to me. It's possible to create new identities even before calling `bt_enable()` the first time, so why should `bt_disable()` touch them?

Reply:
How about a function `bt_id_reset_all()`?

Reply:
But why do we want to do anything about identities to begin with? `bt_disable()` is intended to just shut down the controller and reset any HCI-related host-side states and resources. We're not clearing stuff like the GATT database, unpairing devices, etc, so why would we be resetting or removing local identities?

Reply:
I see, will remove